File: macos/Fliks/State/ServerState.swift
import Foundation

/// Lifecycle state of the Fliks server stack (Postgres + Node).
enum ServerState: Equatable {
    case stopped
    case startingPostgres
    case startingBackend
    case running
    case stopping
    case error(String)

    /// SF Symbol name for the menu bar icon.
    var iconName: String {
        switch self {
        case .stopped:           return "film"
        case .startingPostgres,
             .startingBackend:   return "film.circle"
        case .running:           return "film.fill"
        case .stopping:          return "film.circle"
        case .error:             return "exclamationmark.triangle.fill"
        }
    }

    /// Human-readable status text shown at the top of the menu.
    var displayText: String {
        switch self {
        case .stopped:                return "Fliks is stopped"
        case .startingPostgres:       return "Starting database..."
        case .startingBackend:        return "Starting server..."
        case .running:                return "Fliks is running"
        case .stopping:               return "Stopping..."
        case .error(let message):     return "Error: \(message)"
        }
    }

    var isStarting: Bool {
        switch self {
        case .startingPostgres, .startingBackend: return true
        default: return false
        }
    }
}

File: windows/Fliks.Tray/State/ServerState.cs
namespace Fliks.Tray.State;

internal enum ServerPhase
{
    Stopped,
    StartingPostgres,
    StartingBackend,
    Running,
    Stopping,
    Error,
}

/// <summary>Lifecycle state of the server stack (Postgres + Node).</summary>
internal readonly record struct ServerState(ServerPhase Phase, string? Message = null)
{
    public static readonly ServerState Stopped = new(ServerPhase.Stopped);
    public static readonly ServerState StartingPostgres = new(ServerPhase.StartingPostgres);
    public static readonly ServerState StartingBackend = new(ServerPhase.StartingBackend);
    public static readonly ServerState Running = new(ServerPhase.Running);
    public static readonly ServerState Stopping = new(ServerPhase.Stopping);
    public static ServerState Errored(string message) => new(ServerPhase.Error, message);

    public bool IsStarting =>
        Phase is ServerPhase.StartingPostgres or ServerPhase.StartingBackend;

    public string DisplayText => Phase switch
    {
        ServerPhase.Stopped => "Fliks is stopped",
        ServerPhase.StartingPostgres => "Starting database…",
        ServerPhase.StartingBackend => "Starting server…",
        ServerPhase.Running => "Fliks is running",
        ServerPhase.Stopping => "Stopping…",
        ServerPhase.Error => $"Error: {Message}",
        _ => "",
    };
}
